dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -465,6 +488,42 @@ private boolean isServerError(Exception ex) {
         return false;
     }
 
+    private boolean shouldProcessPush(GitHubSourceCodeRepo gitHubSourceCodeRepo, String repository, String reference, String afterCommit) {
+        // If there's no "after" commit, process the push.
```

Comment:
code comment question, in what situations is there no "after" commit?

Reply:
There will be no "after" commit if we invoke the endpoint manually, and do not include it in the payload.  Doing so would be useful for development and testing, it'll be a pain in the butt to have to locate the ref SHA every time we want to force a push.  I will try to figure out a good place to add an appropriate comment...